atom/browser/net/url_request_async_asar_job.cc

```diff
@@ -22,7 +22,7 @@ void URLRequestAsyncAsarJob::StartAsync(std::unique_ptr<base::Value> options) {
     static_cast<base::DictionaryValue*>(options.get())
         ->GetString("path", &file_path);
   } else if (options->is_string()) {
-    options->GetAsString(&file_path);
+    file_path = options->GetString();
```

Comment:
This can be changed to: 

```
std::string file_path;
if (options->is_dict()) {
  auto path_value = options->FindKeyOfType("path", base::Value::Type::STRING);
  if (path_value)
    file_path = path_value->GetString();
} else if (options->is_string()) {
  file_path = options->GetString();
}

if (file_path.empty()) {
  NotifyStartError(net::URLRequestStatus(net::URLRequestStatus::FAILED,
                                         net::ERR_NOT_IMPLEMENTED));
} else {
  asar::URLRequestAsarJob::Initialize(
      base::CreateSequencedTaskRunnerWithTraits(
          {base::MayBlock(), base::TaskPriority::USER_VISIBLE,
           base::TaskShutdownBehavior::SKIP_ON_SHUTDOWN}),
#if defined(OS_WIN)
      base::FilePath(base::UTF8ToWide(file_path)));
#else
      base::FilePath(file_path));
#endif
  asar::URLRequestAsarJob::Start();
}
```